### PR TITLE
SDIT-2353: ✨ More work on the download

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ services:
       - "--prometheus-disable-collect"
     restart: always
     healthcheck:
-      test: [ 'CMD', 'curl', '-f', 'http://localhost:3001/health' ]
+      test: [ 'CMD', 'curl', '-f', 'http://localhost:3000/health' ]
```
Comment (petergphillips) [inline docker-compose-test.yml:34]: wrong port originally

File: helm_deploy/hmpps-historical-prisoner/values.yaml [hunk 2]
```diff
@@ -51,6 +51,7 @@ generic-service:
     TOKEN_VERIFICATION_ENABLED: 'true'
     AUDIT_SQS_REGION: 'eu-west-2'
     AUDIT_SERVICE_NAME: 'hmpps-historical-prisoner'
+    GOTENBERG_API_URL: 'http://hmpps-historical-prisoner-gotenberg'
```
Comment (petergphillips) [inline helm_deploy/hmpps-historical-prisoner/values.yaml:54]: missing so app wouldn't start - needs to be able to connect to gotenberg pod

File: server/data/gotenbergClient.ts [hunk 3]
```diff
@@ -28,6 +29,7 @@ export default class GotenbergClient {
       .attach('files', Buffer.from(html), 'index.html')
       .attach('files', Buffer.from(headerHtml), 'header.html')
       .attach('files', Buffer.from(footerHtml), 'footer.html')
+      .attach('files', Buffer.from(css), 'app.css')
```
Comment (petergphillips) [inline server/data/gotenbergClient.ts:32]: pass through css instead of getting gotenberg to grab it

File: server/utils/pdfRenderer.ts [hunk 4]
```diff
@@ -30,15 +50,16 @@ export function pdfRenderer(client: GotenbergClient) {
...
-            res.header('Content-Disposition', `inline; filename=${options.filename}`)
-
             client
-              .renderPdfFromHtml(pageHtml, headerHtml, footerHtml, options?.pdfMargins)
-              .then(buffer => res.send(buffer))
-              .catch(reason => {
-                logger.warn(reason)
+              .renderPdfFromHtml(pageHtml, headerHtml, footerHtml, readAssetCss(), options?.pdfMargins)
+              .then(buffer => {
+                res.header('Content-Type', 'application/pdf')
```
Comment (petergphillips) [inline server/utils/pdfRenderer.ts:56]: only do the header writing if we've rendered the pdf.  This then allows the error page to be rendered if the pdf generation fails

File: server/views/partials/pdfLayout.njk [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <html lang="en">
     <head>
         <meta charset="utf-8">
-        <link href="{{ '/assets/css/app.css' | assetMap }}" rel="stylesheet"/>
+        <link href="app.css" rel="stylesheet"/>
```
Comment (petergphillips) [inline server/views/partials/pdfLayout.njk:4]: direct path to `app.css` means that gotenberg uses the passed in css file instead

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**